csharp/src/Ice/InputStream.cs

```diff
@@ -908,50 +945,47 @@ internal Encoding SkipEncapsulation()
             _pos += size - 6;
             if (_pos > _buffer.Count)
             {
-                throw new InvalidDataException("the encapsulation's size extends beyond the end of the frame");
+                throw new InvalidDataException(
+                    $"the encapsulation's size ({size}) extends beyond the end of the frame");
             }
             return encoding;
         }
 
-        private static (Encoding Encoding, int Size) ReadEncapsulationHeader(
-            Encoding encoding, ReadOnlySpan<byte> buffer)
-        {
-            Debug.Assert(encoding == Encoding.V1_1 || encoding == Encoding.V2_0); // for now, only endpoints use 2.0
+        private static int ReadFixedLengthSize(Encoding encoding, ReadOnlySpan<byte> buffer) =>
```

Comment:
Maybe we can use a helper method here and in ReadSize to avoid checking the encoding twice

Reply:
I think it's fine and readable as-is. The encoding check is not that expensive!

Reply:
My idea was that an small method will get a bigger chance of getting inline and will also avoid the call overhead.

If we add the helper we can add `[MethodImpl(MethodImplOptions.AggressiveInlining)] `  to increase the chance for inline it.

Reply:
Yes, I reworked the code a little and added `Read20Size` and `Read11Size` private methods.